nodes/clarify_util.js

```diff
@@ -95,13 +97,17 @@ module.exports = {
     validateString(validationErrors, 'type', type);
```

Comment:
Can we add an (optional) parameter to `validateString` so we can validate the value to be a specific value?

E.g. in this case `["","number","enum"]` where blank string is equivalent to `"number"`.

Reply:
I suppose there are many ways to do this. Different method names might be the best. e.g.:
- `validateString(validationErrors, name , value)`
- `validateStringEnum(validationErrors, name , value, allowed)`
- `validateStringPattern(validationErrors, name, value, regex)`

Reply:
Updated the PR with an extra commit. Didn't implement `validateStringPattern` since we don't have any use cases for it.

Reply:
The use case for validateStringPattern was actually just a more general form of the  validateStringRFC3339.

Not that you need to change it; `validateStringRFC3339` is fine.